### DISCUSSION TITLE
Port question client side.

### DISCUSSION THREAD

**Discussion by Nawaki89** [rustdesk/rustdesk-server-pro#389] [Q&A]
I'v been rolling out Rustdesk on a lot of clients PC but they have all been hosted on my "own" network I control but I havn't encounter any problems with the client connecting out.    now i'm installing on a PC that is on a network someone els is managing and the client can't connect out seems to be very strict firewall on this network.     so now they are ask me what ports I need open but the client uses random port each attempt it connects out? if i'v done my research.    so what ports should they open?

**Comment by rustdesk** [comment]
Outbound ports on client: https://rustdesk.com/docs/en/self-host/#ports-required, if the `21116/UDP` port is an issue for you, you can disable `UDP`, https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#disable-udp.

You do not need to worry about the random inbound port for `P2P`; if `P2P` fails, it will automatically switch to connecting via a relay. Additionally, you can completely disable this random inbound port by setting [`always use relay`](https://github.com/rustdesk/rustdesk/wiki/FAQ#you-can-force-to-use-relay-for-all-connections-in-web-console-of-pro) or turning on the option below.

![image](https://github.com/user-attachments/assets/18cea1de-3635-4b3a-aff4-7a27e149c38f)


If you have only `443/80` ports allowed, you can use `http/https` proxy. You need to set up additional `http/https` proxy server yourself.

https://github.com/rustdesk/rustdesk/wiki/Set-up-http-proxy-server

**Reply by Nawaki89** [reply]
Yeah the server is solid and uses https already got as I mentioned a lot of clients connected to it already.   It's just this "site" that won't connect even tho the firewall is disable on the PC for testing purposes.  So my trail let me to believe it's the outbound port the client uses to establish connection to the relay.  Iv already opt in on the server side to use relay always. 

I will try generation a new custom client with the option enable you mentioned and give it a try.   Should maybe see if I can ping port 443 on the pc to server so that port isn't closed? If that is used for connection purposes since the server is setup with domain and TLS?
